src/group_theory/torsion.lean

```diff
@@ -73,15 +73,31 @@ lemma is_torsion.subgroup (tG : is_torsion G) (H : subgroup G) : is_torsion H :=
 
 /-- Quotient groups of torsion groups are torsion groups. -/
 @[to_additive "Quotient groups of additive torsion groups are additive torsion groups."]
-lemma is_torsion.quotient_group [nN : N.normal] (tG : is_torsion G) : is_torsion (G ⧸ N) :=
+lemma is_torsion.quotient_group [N.normal] (tG : is_torsion G) : is_torsion (G ⧸ N) :=
 λ h, quotient_group.induction_on' h $ λ g, (tG g).quotient N g
 
+/-- Torsion groups are closed under extensions. -/
+@[to_additive add_is_torsion.extension_closed
+  "Additive torsion groups are closed under extensions."]
+lemma is_torsion.extension_closed (tN : is_torsion N) [N.normal] (tGN : is_torsion (G ⧸ N)) :
```

Comment:
Could you please generalize this to surjective maps `f : G → H` and `N = ker f`? Even better might be to also turn `N → G` into an arbitrary group hom, whose range is the kernel of `f`.
That way the lemma would be easier to apply in a situation that is morally a quotient, but not defeq to one.

Reply:
I've given this a shot I think -- apologies if I've misunderstood the suggestion, especially since I think the surjective assumption was only required in a different spot. Let me know if I've indeed put nonsense up here now. And thanks as usual for the pointer.